www/public/index.html

```diff
@@ -41,8 +55,241 @@ <h1>Work in Progress</h1>
+    at least as fast as C.</p>
+
+  <p>When benchmarking Roc code against similarly-optimized <a href="https://go.dev">Go</a>,
+    <a href="https://www.swift.com/">Swift</a>, <a href="https://www.java.com">Java</a>,
+    <a href="https://learn.microsoft.com/en-us/dotnet/csharp">C#</a>, and
+    <a href="https://www.ecma-international.org/publications-and-standards/standards/ecma-262">JavaScript</a>,
+    we generally aim for Roc to outperform all of those languages. Outperforming systems
+    languages like Rust, Zig, C, D, and C++ is a non-goal, as is outperforming research languages
+    that see little or no use in industry. (Presumably there will always be certain specific
+    benchmarks where some popular slower-than-C languages outperform Roc, but the goal is to
+    usually be at the front of that pack.)</p>
+
+  <h4>Current progress</h4>
+
+  <p>Progress towards this performance goal is already quite far along.</p>
+
+  <p>Since Roc already uses unboxed data structures that get monomorphized into LLVM—just like Rust,
+    Zig, and C++ do—in many cases Roc code already compiles to the same machine instructions that the
+    equivalent code written in one of these systems languages would. Something we do regularly is
+    to compare the LLVM instructions generated by Roc's compiler and by these systems languages'
+    compilers, to check whether we're generating equivalent instructions.</p>
+
+  <p>That said, there are also cases where Roc has strictly more runtime overhead than languages
+    like C, C++, Zig, and Rust do. The most significant is automatic memory management, which Roc
+    implements using automatic reference counting. Static reference count optimizations like
+    elision and reuse (thanks to Morphic and
+    <a href="https://www.microsoft.com/en-us/research/publication/perceus-garbage-free-reference-counting-with-reuse/">Perceus</a>)
+    improve things, but significant runtime overhead remains.</p>
+
+  <p>Eliminating this overhead altogether would require sacrificing other design goals
+  (e.g. it would require introducing memory-unsafe operations, or compile-time lifetime errors),
+  and there isn't much overhead left to remove outside of automatic memory management. For example,
+  smaller sources of overhead include mandatory array bounds checks, disallowing cyclic references
+  (which rules out a certain niche of efficient graph data structures), and automatic opportunistic
+  in-place mutation instead of direct mutation. Even if all of these sources of overhead were
+  completely eliminated, it seems unlikely that typical Roc programs would see a particularly big
+  performance boost.</p>
+
+<p>Overall, we expect Roc's performance in the use cases mentioned above (servers, CLIs, GUIs, etc.)
+   to be about the same as the equivalent C++ code would be, if all that C++ code
+   (including dependencies other than the stdlib) were written in a restricted subset of C++
```

Comment:
Probably including the stdlib, right? We must use reference counts in Roc's standard library as well.